src/vizier/services/metadata/controllers/k8smeta/k8s_metadata_handler.go

```diff
@@ -938,13 +939,16 @@ func getResourceUpdateFromPod(pod *metadatapb.Pod, uv int64) *metadatapb.Resourc
 		hostname = pod.Spec.Hostname
 	}
 
+	podLabels, _ := json.Marshal(pod.Metadata.Labels)
+
```

Comment:
you need a null check here. if `pod.Metadata.Labels` is null, then `podLabels` will be the literal string `"null"`

Reply:
Fixed with https://github.com/pixie-io/pixie/pull/468/commits/233860db09eecc9fb6827bae2eef3b6eb8f48926